File: pytools/examples/csharp/server/Protocol/MsgSceneVector3.cs
using System.Collections;
using System.Collections.Generic;


public class MsgSceneVector3
{
	private short _x;
	private short _y;
	private short _z;


	public Packet Encode()
	{
		Packet packet = new Packet();
		packet.WriteShort(this._x);
		packet.WriteShort(this._y);
		packet.WriteShort(this._z);
		return packet;
	}

	public MsgSceneVector3()
	{
	}

	public MsgSceneVector3(Packet packet)
	{
		this._x = packet.ReadShort();
		this._y = packet.ReadShort();
		this._z = packet.ReadShort();
	}

	public List<byte> GetBuffer()
	{
		return this.Encode().GetBuffer();
	}


	public short x
	{
		get { return this._x; }
		set { this._x = value; }
	}

	public short y
	{
		get { return this._y; }
		set { this._y = value; }
	}

	public short z
	{
		get { return this._z; }
		set { this._z = value; }
	}

}

File: pytools/examples/cpp/server/protocol/MsgSceneVector3.cpp
#ifndef _MSG_SCENE_VECTOR_3_
#define _MSG_SCENE_VECTOR_3_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class MsgSceneVector3
{
private:
	I16 m_x;
	I16 m_y;
	I16 m_z;


public:
	BYTE* Encode(U32* len)
	{
		Packet packet;
		packet.WriteI16(m_x);
		packet.WriteI16(m_y);
		packet.WriteI16(m_z);
		return packet.ReadBytes(len);
	}

	MsgSceneVector3()
	{

	}

	MsgSceneVector3(Packet* packet)
	{
		m_x = packet->ReadI16();
		m_y = packet->ReadI16();
		m_z = packet->ReadI16();
	}

	BYTE* GetBytes(U32* len)
	{
		return Encode(len);
	}


	void SetX(I16 x)
	{
		m_x = x;
	}
	I16 GetX()
	{
		return m_x;
	}

	void SetY(I16 y)
	{
		m_y = y;
	}
	I16 GetY()
	{
		return m_y;
	}

	void SetZ(I16 z)
	{
		m_z = z;
	}
	I16 GetZ()
	{
		return m_z;
	}

};

#endif
